### ISSUE TITLE
RPC Listen

### ISSUE THREAD

**Issue by croll83** [LN-Zap/zap-tutorials#1]
Hi Jack,

found an issue following your tutorial.

--rpclisten=0.0.0.0:10009

It doesn't work on my lnd node. if I bind on all interface it only listen to tcp6 not tcp4. If I bind on localhost or remote interface it works.

Do we need to expose RPC over internet? how secure it is? Is there any other way to use iOS ZAP? neutrino?

Thanks,

Marco

**Comment by giusr** [comment]
The node IP provided refuses all connections, also as @croll83 mentioned LND only listens on ipv6.

**Comment by mrfelton** [comment]
> Do we need to expose RPC over internet?

Yes, if you want to be able to connect Zap iOS or Zap Desktop to your node then the gRPC interface needs to be exposed over the internet as that is how we communicate with it.

> how secure it is?

Exposing the interface by itself is safe enough. It can't be used for anything without your access credentials.

> Is there any other way to use iOS ZAP? neutrino?

Neutrino is not yet up to the task, but it's getting there.

**Comment by dell1503** [comment]
Hi there,
@mrfelton 
I had the same issue. it shows me only ipv6 in netstat.
Can't connect from external to the LND Node. From the LAN it works.
When connecting from External the connections appears in netstat:
[netstat](https://ibb.co/4snPJBh)

But the App says no connection to the Server.

The Ports, 10009 and 9735 are both exposed to the internet. (TCP & UDP)
I run LND with the following config:
(Have a local Bitcoinserver running, mainnet)

`[Application Options]'
; set external IP if not using NAT
externalip=94.XX.XX.XX
; set node alias (seen in explorers)
alias=AlpaLND

tlsextraip=94.XX.XX.XX
rpclisten=0.0.0.0:10009

[Bitcoin]
bitcoin.active=1
bitcoin.mainnet=1
bitcoin.node=bitcoind

[Bitcoind]
bitcoind.rpcuser=XXX
bitcoind.rpcpass=XXX
bitcoind.zmqpubrawblock=tcp://127.0.0.1:28332
bitcoind.zmqpubrawtx=tcp://127.0.0.1:28333

[autopilot]

; If the autopilot agent should be active or not. The autopilot agent will
; attempt to automatically open up channels to put your node in an advantageous
; position within the network graph.
autopilot.active=1`

Node is online:
https://1ml.com/node/0245eba55d18d41008f0b08b63252f73f84c020f9976091d6248d1722915f26cfc
LND runs on Ubuntu Server 18.04.

Thanks fo a hint!

**Comment by missaghi** [comment]
Instead of exposing RPC to internet you can use a VPN. Tutorial PR pending.
